packages/cli/src/build.ts

```diff
@@ -0,0 +1,63 @@
+import * as esbuild from 'esbuild';
+import fs from 'fs';
+import path from 'path';
+
+import { resolveFile, readIntegrationManifest } from './manifest';
+
+const GITBOOK_OUTPUT_DIR = '.gitbook';
+
+interface BuildOutput {
+    /** path where the final script is outputted */
+    path: string;
+    /** the text content of built script */
+    script: string;
+    /** resolved manifest config used during the process */
+    manifest: Awaited<ReturnType<typeof readIntegrationManifest>>;
+}
+
+/**
+ * Build the script into a single worker definition.
+ */
+export async function buildScriptFromManifest(manifestSpecPath: string): Promise<BuildOutput> {
+    const manifest = await readIntegrationManifest(manifestSpecPath);
+    /**
+     * Resolve the input and output file paths relatve to the manifest file.
+     */
+    const inputFilePath = resolveFile(manifestSpecPath, manifest.script);
+    const outputFilePath = resolveFile(
+        manifestSpecPath,
+        path.join(GITBOOK_OUTPUT_DIR, 'script.js')
+    );
+
+    await esbuild.build({
+        platform: 'neutral',
+        entryPoints: [inputFilePath],
+        outfile: outputFilePath,
+        bundle: true,
+        minify: true,
```

Comment:
In development, we probably don't want this? or we want sourcemap?

Reply:
Yes, I didn't give thought to this part but probably something along the lines of what you said